pyzwaver/driver.py

```diff
@@ -412,6 +412,10 @@ def _DriverReceivingThread(self):
                 # Does this help?
                 # TODO: analyze
                 time.sleep(0.01)
+                # What's the reason for the sleep() command? 
```

Comment:
Sadly this code is indeed fragile.
At least with my aeotech zwave stick 
the stick can get confused when commands are send to quickly.
This is the reason for some of the sleep statements.

A few nits:
* Can mention the type of zwave stick you observed this with.
* reformat with standard python formatter
  (line length is 80, continue on separate  line)

Reply:
Same - Aeotec Z-Stick Gen5. The driver code seems in general not overly resilient. I think you could probably crash it with malformed packages (haven't really checked though) and I've observed multiple race conditions when the zmessage.inflight field is invalidated by a separate thread while still being accessed. No issue if it's the timeout-thread crashing but the above made the DriverReceiving thread crash. There should probably be some locking/synchronisation around it.

Reply:
you are right, there needs to be some synchronization - will look into it over Christmas.
(don't forget about the reformatting.)

Reply:
I just pushed an update that cleans up the in-flight message handling code.
Even if does not fix all the races the code should be much easier to evolve now.